src/metabase/pulse/render/body.clj

```diff
@@ -213,14 +213,28 @@
   {:arglists '([chart-type render-type timezone-id card dashcard data])}
   (fn [chart-type _ _ _ _ _] chart-type))
 
+(defn- filter-columns [filtered-column-names rows]
+  (let [filtered-headers-idx
+        (keep-indexed (fn [idx header-cell] (when (some #{(h header-cell)} filtered-column-names) idx))
+                      (:row (first rows)))
+        keep-filtered-idx #(keep-indexed (fn [idx header-call] (when (some #{idx} filtered-headers-idx) header-call)) %)
+        filtered-rows (map #(update % :row keep-filtered-idx) rows)]
+    filtered-rows))
+
 (s/defmethod render :table :- common/RenderedPulseCard
   [_ render-type timezone-id :- (s/maybe s/Str) card dashcard {:keys [rows viz-settings] :as data}]
   (let [viz-settings (merge viz-settings (:visualization_settings dashcard))
+        filter-columns? (some? (:table.columns viz-settings))
+        filtered-columns-names (if filter-columns?
+                                 (mapv :name (filter :enabled (:table.columns viz-settings)))
+                                 (mapv :name (:cols data)))
```

Comment:
Tested locally and it didn't work for me, I think because you're not using the correct keys for pulling data out of `viz-settings`. They should be namespaced with `mb.viz` like `::mb.viz/table-columns`.

Reply:
Tested with `::mb.viz/table-columns` and didn't work for me, but it does work for `:table.columns`. What does your `viz-settings` look like? It doesn't show `::mb.viz/table-columns ` for me.

```{:table.pivot_column QUANTITY, :table.cell_column ID, :metabase.shared.models.visualization-settings/column-settings nil, :metabase.shared.models.visualization-settings/global-column-settings {:type/Temporal {:metabase.shared.models.visualization-settings/date-style D/M/YYYY, :metabase.shared.models.visualization-settings/date-separator -, :metabase.shared.models.visualization-settings/date-abbreviate false}, :type/Number {:metabase.shared.models.visualization-settings/number-separators ,.}}, :table.columns [{:name ID, :fieldRef [:field ID {:base-type :type/BigInteger}], :enabled true} {:name USER_ID, :fieldRef [:field USER_ID {:base-type :type/Integer}], :enabled true} {:name SUBTOTAL, :fieldRef [:field SUBTOTAL {:base-type :type/Float}], :enabled true} {:name TAX, :fieldRef [:field TAX {:base-type :type/Float}], :enabled true} {:name CREATED_AT, :fieldRef [:field CREATED_AT {:base-type :type/DateTime}], :enabled true} {:name TOTAL, :fieldRef [:field TOTAL {:base-type :type/Float}], :enabled false} {:name DISCOUNT, :fieldRef [:field DISCOUNT {:base-type :type/Float}], :enabled false} {:name PRODUCT_ID, :fieldRef [:field PRODUCT_ID {:base-type :type/Integer}], :enabled false} {:name QUANTITY, :fieldRef [:field QUANTITY {:base-type :type/Integer}], :enabled false}]}```

Reply:
Here's what I see in `viz-settings` in `render`:

```
{:table.pivot_column "QUANTITY",
 :table.cell_column "ID",
 :metabase.shared.models.visualization-settings/table-columns
 [#:metabase.shared.models.visualization-settings{:table-column-name "ID",
                                                  :table-column-field-ref [:field "ID" {:base-type :type/BigInteger}],
                                                  :table-column-enabled false}
  #:metabase.shared.models.visualization-settings{:table-column-name "USER_ID",
                                                  :table-column-field-ref
                                                  [:field "USER_ID" {:base-type :type/Integer}],
                                                  :table-column-enabled false}
  #:metabase.shared.models.visualization-settings{:table-column-name "PRODUCT_ID",
                                                  :table-column-field-ref
                                                  [:field "PRODUCT_ID" {:base-type :type/Integer}],
                                                  :table-column-enabled false}
  #:metabase.shared.models.visualization-settings{:table-column-name "SUBTOTAL",
                                                  :table-column-field-ref [:field "SUBTOTAL" {:base-type :type/Float}],
                                                  :table-column-enabled false}
  #:metabase.shared.models.visualization-settings{:table-column-name "TAX",
                                                  :table-column-field-ref [:field "TAX" {:base-type :type/Float}],
                                                  :table-column-enabled false}
  #:metabase.shared.models.visualization-settings{:table-column-name "TOTAL",
                                                  :table-column-field-ref [:field "TOTAL" {:base-type :type/Float}],
                                                  :table-column-enabled false}
  #:metabase.shared.models.visualization-settings{:table-column-name "DISCOUNT",
                                                  :table-column-field-ref [:field "DISCOUNT" {:base-type :type/Float}],
                                                  :table-column-enabled true}
  #:metabase.shared.models.visualization-settings{:table-column-name "CREATED_AT",
                                                  :table-column-field-ref
                                                  [:field "CREATED_AT" {:base-type :type/DateTime}],
                                                  :table-column-enabled true}
  #:metabase.shared.models.visualization-settings{:table-column-name "QUANTITY",
                                                  :table-column-field-ref
                                                  [:field "QUANTITY" {:base-type :type/Integer}],
                                                  :table-column-enabled true}],
 :metabase.shared.models.visualization-settings/column-settings nil,
 :metabase.shared.models.visualization-settings/global-column-settings {}}
```